tools/autograd/derivatives.yaml

```diff
@@ -1939,9 +1939,9 @@
   grad_output: log_sigmoid_backward(grad, self, buffer)
   self: log_sigmoid_double_backward(grad * grad_output, self)
 
-- name: _log_softmax_backward_data(Tensor grad_output, Tensor output, int dim) -> Tensor
+- name: _log_softmax_backward_data(Tensor grad_output, Tensor output, int dim, ScalarType input_dtype) -> Tensor
   grad_output: grad.to(output.dtype()) - (grad.to(output.dtype()) * output.exp()).sum(dim, true)
-  output: log_softmax_double_backward(grad.to(output.dtype()), grad_output, dim, output).to(output.dtype())
+  output: log_softmax_double_backward(grad.to(output.dtype()), grad_output, dim, output).to(input_dtype)
```

Comment:
should it be `output.dtype()` here, `output` grad should have the same dtype as `output`?

Reply:
Whoops, good catch